lib/pagy/extras/headers.rb

```diff
@@ -0,0 +1,35 @@
+# See the Pagy documentation: https://ddnexus.github.io/pagy/extras/headers
+# encoding: utf-8
+# frozen_string_literal: true
+
+class Pagy
+  # Add specialized backend methods to add pagination response headers
+  module Backend ; private
+
+    include Helpers
+
+    def pagy_headers_merge(pagy)
+      response.headers.merge!(pagy_headers(pagy))
+    end
+
+    def pagy_headers(pagy)
+      hash = pagy_headers_hash(pagy)
+      { 'Links'    => hash[:links].map{|rel, link| %(<#{link}>; rel="#{rel}")}.join(', ') }.tap do |h|
+        h['Items'] = h['Per-Page'] = hash[:items]
+        h['Count'] = h['Total']    = hash[:count] if hash.key?(:count)
+      end
+    end
+
+    def pagy_headers_hash(pagy)
+      countless = defined?(Pagy::Countless) && pagy.is_a?(Pagy::Countless)
+      rels      = { first: 1, prev: pagy.prev, next: pagy.next }.tap{ |r| r[:last] = pagy.last unless countless }
+      a, b      = pagy_url_for(Frontend::MARKER, pagy, :url).split(Frontend::MARKER, 2)
```

Comment:
got some more descriptive names than a/b ?

Reply:
like `part_a` and `part_b`? :smile:

Reply:
was hoping they have a meaning like url/query "MARKER" is also vague :D

Reply:
well, MARKER means... it's just a MARKER :smile: ... used to split the string into 2 parts and wrap them around the page number. That was faster than using `sub` to replace MARKER with the page number (at least in `pagy_link_proc`), but the 2 parts are just 2 strings without any logical meaning. It's difficult to give them a meaningful name.